lib/drives/pages/drive_chat_page.dart

```diff
@@ -16,107 +16,110 @@ class DriveChatPage extends StatefulWidget {
-            _riderList(riders),
-          ];
-          widgets.addAll(riderColumn);
-        }
-        widgets.add(const SizedBox(height: 10.0));
-      }
-    } else {
-      widgets.add(const SizedBox(height: 10));
-      widgets.add(const Center(child: CircularProgressIndicator()));
-    }
     return Scaffold(
       appBar: AppBar(
         title: Text(S.of(context).pageDriveChatTitle),
       ),
-      body: RefreshIndicator(
-        onRefresh: loadDrive,
-        child: ListView.separated(
-          itemCount: widgets.length,
-          itemBuilder: (context, index) {
-            return widgets[index];
-          },
-          separatorBuilder: (BuildContext context, int index) {
-            return const SizedBox(height: 10);
-          },
-        ),
+      body: StreamBuilder<List<Message>>(
+        stream: _messagesStream,
+        builder: (context, snapshot) {
+          if (snapshot.hasData) {
+            Set<Ride> approvedRides = widget.drive.approvedRides!.toSet();
+            for (Message message in snapshot.data!) {
+              Ride ride = approvedRides.firstWhere((element) => element.id == message.rideId);
+              if (ride.messages!.contains(message)) {
+                ride.messages!.remove(message);
+              }
+              ride.messages!.add(message);
+            }
+            List<Widget> widgets = _buildChatWidgets(approvedRides);
+            return ListView.separated(
+              itemCount: widgets.length,
+              itemBuilder: (context, index) {
+                return widgets[index];
+              },
+              separatorBuilder: (BuildContext context, int index) {
+                return const SizedBox(height: 10);
+              },
+            );
+          } else {
+            return const Center(
+              child: CircularProgressIndicator(),
+            );
+          }
+        },
       ),
     );
   }
 
-  Widget _riderList(Set<Profile> riders) {
-    Widget ridersColumn = Container();
-    if (riders.isNotEmpty) {
-      ridersColumn = Column(
-        children: List.generate(
-          riders.length,
-          (index) => Padding(
-            padding: const EdgeInsets.symmetric(horizontal: 5.0, vertical: 5.0),
-            child: ProfileWidget(
-              riders.elementAt(index),
-              //TODO go to chat
-              onTap: () {},
-              actionWidget: const Icon(
-                Icons.chat,
-                color: Colors.black,
-                size: 36.0,
-              ),
-            ),
-          ),
+  Widget _buildChatWidget(Ride ride) {
+    ride.messages!.sort((a, b) => b.createdAt!.compareTo(a.createdAt!));
+    Message? lastMessage = ride.messages!.isEmpty ? null : ride.messages!.first;
+    return Card(
+      child: InkWell(
+        child: ListTile(
+          leading: Avatar(ride.rider!),
+          title: Text(ride.rider!.username),
+          subtitle: lastMessage == null ? null : Text(lastMessage.content),
```

Comment:
Please limit the length of this

![image](https://user-images.githubusercontent.com/69083092/212776867-d53852a3-eb67-45c3-b73e-ec3b31dbb890.png)

Reply:
thanks for the hint i wanted to do that but forgot i guess.